pytorch_lightning/trainer/trainer.py

```diff
@@ -807,6 +801,15 @@ def tune(
 
         return result
 
+    @property
+    def ckpt_path(self) -> Optional[str]:
+        if self.state.fn == TrainerFn.VALIDATING:
+            return self.validated_ckpt_path
+        if self.state.fn == TrainerFn.TESTING:
+            return self.tested_ckpt_path
+        if self.state.fn == TrainerFn.PREDICTING:
+            return self.predicted_ckpt_path
+
```

Comment:
This might not be the most descriptive thing ever, so if anyone has suggestions let me me know!

Reply:
LGTM, but make it protected `_ckpt_path` as it will only work if accessed inside the trainer flow. Will return `None` if the `Trainer` is not running

Reply:
Also, move it to `TrainerProperties`